### PR TITLE
feat(store-binary-cid): Store binary CID/retrieve in Base32

### LINKED ISSUE #787: Message RPC returns IPFS CIDv1s as String base32
Our output of IPFS CIDs is not properly sanitized and serialized.
Outcome: only output IPFS CIDv1 in `base32` encoding

Partially dependent on #786

Breaking Change:
- Before: `MessageResponse` returned CID of type `Option<Vec<u8>>` which could have been anything
- After: `MessageResponse` returns CID of type `Option<string>` which is the cidv1 in `base32`

Related:
- #738 
- #786 
- CID Spec: https://github.com/multiformats/cid
- CID Inspector: https://cid.ipfs.tech
- JS for multiformats https://github.com/multiformats/js-multiformats/blob/master/src/cid.js

## Acceptance Criteria
- [x] `add_ipfs_message` with:
    - [x] string: `bafybeihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku`
    - [x] stores payload as hex: `0x01701220e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`
    - [x] Returns `get_messages_by_schema_id` base32 string of:`bafybeihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku`
- [x] Better documentation around the IPFS response as always `CIDv1`
    - [x] https://libertydsnp.github.io/frequency/common_primitives/messages/struct.MessageResponse.html#structfield.cid

### PR DESCRIPTION
**Pull request by JoeCap08055**
# Goal
- Always store IPFS CID as binary on-chain to reduce payload storage footprint
- Always return retrieved CID using Base32 encoding

Closes #787 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [x] Tests added
- [ ] Benchmarks added
- [x] Weights updated

### CODE REVIEW

File: pallets/messages/src/lib.rs [hunk 1]
```diff
@@ -221,18 +221,16 @@ pub mod pallet {
...
 			#[pallet::compact] schema_id: SchemaId,
 			cid: Vec<u8>,
 			#[pallet::compact] payload_length: u32,
 		) -> DispatchResult {
 			let provider_key = ensure_signed(origin)?;
-			// validate_cid returns the binary CID representation. We'll use that in an upcoming PR, not now.
-			// For now we just store the input CID in the payload.
-			Self::validate_cid(&cid)?;
-			let payload_tuple: OffchainPayloadType = (cid, payload_length);
+			let cid_binary = Self::validate_cid(&cid)?;
```
Comment (aramikm) [inline pallets/messages/src/lib.rs:232]: Quick question: do we know in which cases `cid` will be different from `cid_binary`?

**Review by aramikm — APPROVED** [review]
Look good! 

download (1)

File: pallets/messages/src/lib.rs [hunk 2]
```diff
@@ -221,18 +221,16 @@ pub mod pallet {
 		/// * [`Error::UnsupportedCidVersion`] - CID version is not supported (V0)
 		/// * [`Error::InvalidCid`] - Unable to parse provided CID
 		///
-		#[pallet::weight(T::WeightInfo::add_ipfs_message(cid.len() as u32))]
+		#[pallet::weight(T::WeightInfo::add_ipfs_message())]
```
Comment (rlaferla) [inline pallets/messages/src/lib.rs:224]: 🎉 yes! 🎉

**Review by rlaferla — APPROVED** [review]
L👀ks good!

File: pallets/messages/src/lib.rs (hunk 1 above)
Comment (JoeCap08055) [reply pallets/messages/src/lib.rs:232]: cid will always be an encoded string (base32, base64, etc). (We will only accept string-encoded CIDs). Binary will always be the decoded version of the input string. So they will always be different.

**Pull request merged**